nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -42,8 +42,127 @@ let
+    description = "mount host path into container";
+  };
+
+  fromBindMounts = mapAttrsToList (dest: { source, mountOption, ... }:
+    if mountOption != "" then "${source}:${dest}:${mountOption}"
+    else if source != "" then "${source}:${dest}"
+    else toString dest);
+
+  /*fromBindMounts = mountType: attrs: concatStringsSep "\n"
+    (mapAttrsToList
+      (dest: { source, mountOption, ... }:
+        if mountOption != "" then "${mountType}=${source}:${dest}:${mountOption}"
+        else if source != "" then "${mountType}=${source}:${dest}"
+        else "${mountType}=${dest}")
+      attrs);
+  */
+  fromTmpfsMounts = map (dest:
+    if builtins.isString dest then toString dest
+    else "${dest.path}:${dest.mountOption}");
+
+  /*fromTmpfsMounts = mountType: list: concatStringsSep "\n"
+    (map (dest:
+      if builtins.isString dest then "${mountType}=${dest}"
+      else "${mountType}=${dest.path}:${dest.mountOption}"
+    ) list);*/
+
+  /* fromInaccessibles = mountType: list: concatStringsSep "\n"
+    (map (path: "${mountType}=${path}") list); */
+  fromInaccessibles = map toString;
+
+  overlayMounts = mkOption {
+    type = with types; attrsOf (listOf (either str path));
+    default = {};
+    example = { "/var" = [ "+/readonly-var" "" ]; };
+    description = "Mount host path into the container.";
+  };
+
+  /* fromOverlayMounts = mountType: attrs: concatStringsSep "\n"
+    (mapAttrsToList
+      (dest: sources:
+        if length sources == 0 then throw "Overlay mount source path list is empty"
+        else "${mountType}=${concatStringsSep ":" sources}:${dest}")
+      attrs); */
+  fromOverlayMounts = mapAttrsToList (dest: sources:
+    if length sources == 0 then throw "Overlay mount source path list is empty"
+    else "${concatStringsSep ":" sources}:${dest}");
+
+  fromVeths = mapAttrsToList (container: host: "${host}:${container}");
+
   instanceOptions = {
-    options = sharedOptions // {
+    options = {
+
+      environment = mkOption {
+        default = {};
+        example = { PATH = "/run/current-system/sw/bin"; };
+        type = with types; attrsOf (nullOr (either str (either path package)));
+        description = "Environment variables passed to the container's init.";
+      };
+
+      bind = bindMounts;
+      bindReadOnly = bindMounts;
+
+      temporaryFileSystem = mkOption {
+        type = with types; listOf
+          (oneOf
+            [ str
+              (submodule {
+                options = {
+                  path = mkOption {
+                    type = either str path;
+                    default = "";
+                    description = "mount destination";
+                  };
+                  mountOption = mkOption {
+                    type = str;
+                    default = "";
+                    description = "mount options. See https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html#Mount%20Options for more info.";
```

Comment:
Ideally, the descriptions should be sentences (with capital letter at the start).

Reply:
Nice catch